docs/design/values.md

```diff
@@ -623,6 +626,121 @@ The model of initialization of returns also facilitates the use of
 [`returned var` declarations](control_flow/return.md#returned-var). These
 directly observe the storage provided for initialization of a function's return.
 
+### Expression forms
+
+We typically treat the category and type of an expression as independent
+properties. However, in some cases they can become entangled, and we need deal
+with them as an integrated whole. The _form_ of an expression captures all of
```

Comment:
I think it is more about capturing all of them rather than them being particularly entangled.